### PR TITLE
[TorchFX] Constant folding

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

Constant folding is enabled by default in TorchFX backend

### Reason for changes

To align quantizers placement between OV and TorchFX

### Related tickets

#2766 

### Tests

* test_constant_folding
*  test_constant_folding_with_constraints
* test_models.py references are updated
* post_training_quantization/535/ - finished successfully

### CODE REVIEW

File: nncf/experimental/torch/fx/constant_folding.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+        # We need to do this weird thing because in cases where flattened_inputs
+        # contains a ScriptObject, equality checking results in a type error if
+        # the types are different.
+        if any(
+            type(self.unknown_value) is type(input_) and self.unknown_value == input_ for input_ in flattened_inputs
+        ):
+            return self.unknown_value
+
+        # TODO - fix errors with this
```
Comment (alexsu52) [inline nncf/experimental/torch/fx/constant_folding.py:170]: You should remove TODO because they don't align with the style guide.

Comment (daniil-lyakhov) [reply nncf/experimental/torch/fx/constant_folding.py]: Done

**Review by anzr299 — APPROVED** [review]

File: nncf/experimental/torch/fx/transformations.py [hunk 2]
```diff
@@ -802,12 +803,26 @@ def apply_quantization_transformations(model: torch.fx.GraphModule) -> None:
     # to make it easier for algorithms to work
     # with the target graph BatchNorm operations
     # are being fused
+    fold_constant_except_qdq(model)
```
Comment (anzr299) [inline nncf/experimental/torch/fx/transformations.py:806]: Can constant folding also be applied for weights compression?

Comment (daniil-lyakhov) [reply nncf/experimental/torch/fx/transformations.py:806]: Can you provide an use case / model to justify this change?

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**